src/transforms/ppmt.jl

```diff
@@ -0,0 +1,171 @@
+        end
+    end
+    return Float64[1 *(iₘₐₓ==i) for i = 1:q]
+end
+
+function max_diagonal(Z,J)
+    #Given the maximizer principal direction, finds the diagonal direction that #maximizes I.
+    ᾱ = max_principal(Z,J)
+    l = length(ᾱ)
+    I₀ = projection_index(Z,ᾱ,J)
+    for i in 1:l
+        eᵢ = Float64[1*(j==i) for j=1:l]
+        f₊ = projection_index(Z, (1/√(2+2ᾱ[i]))*(ᾱ + eᵢ), J)
+        f₋ = projection_index(Z, (1/√(2-2ᾱ[i]))*(ᾱ - eᵢ), J) * (ᾱ[i] != 1)
+        f = (f₊>f₋)*f₊ + (f₊<=f₋)*f₋
+        s = (f₊>f₋)*1 + (f₊<=f₋)*(-1)
+        if f > projection_index(Z,ᾱ,J)
+            ᾱ = (1/√(2+2s*ᾱ[i]))*(ᾱ + s*eᵢ)
+        end
+    end
+    if projection_index(Z,ᾱ,J) >= I₀
+        return ᾱ
+    else
+        return max_principal(Z,J)
+    end
+end
+
+function optimizer(Z,J)
+    #Given the maximizer diagonal direction, finds a local maximum for I.
+    α₀ = max_diagonal(Z,J)
+    f(α) = -projection_index(Z,α,J)
+    op = optimize(f,α₀)
+    Optim.minimizer(op)
+end
+
+function structure_removal(Z,α)
+    #Transforms the cdf of a convex combination of r.v. to a standard normal cdf.
+    l = length(α)
+    U = qr(hcat(α,rand(l,l-1)))
+    if norm(diag(U.R))<1e-5 #Considers if the randomly created matrix is singular.
+        return print("Congrats: you have hit the almost impossible chance of drawing $l-1 random l.d. vectors!")
+    end
+    T = Z * U.Q
+    cdf = EmpiricalCDF(T[:,1])
+    sort!(cdf)
+    T[:,1] = [quantile(Normal(), q) for q in cdf(T[:,1]) .- 1e-5]
+    return T*U.Q'
+end
+
+function gaussian_projection_matrix(N,q,J,k)
+    #Gives the array that will be used to stop the loop of structure removal by computing the projection index on k samples of multinomial distribution (size of sample kxn) along q random directions.
+    I = ones(k,q)
+    d = MvNormal(zeros(q), Diagonal(ones(q)))
+    I = Matrix{Float64}(undef,k,q)
+    Q = qr(rand(q,q)).Q
+    for i = 1:k
+        S = rand(d,N)'
+        for j = 1:q
+            I[i,j] = projection_index(S,Q[:,j],J)
+        end
+    end
+    I, Q
+end
+
+function stopping_criteria(pi,Z,I,Q,J)
+    #Computes the projection index of each loop dataset (with possibly modified structure) along q random directions and compares with the array given by gaussian_projection_matrix
+    k, q = size(I)
+    switch = true
+    for i = 1:q
+        b = projection_index(Z,Q[:,i],J)
+        switch *= quantile(I[1:k-1,i],pi) > b
+    end
+    return !switch
+end
+
+function ppmt(D,pi,J,k)
+    #Generates the loop for the structure removal process
+    Z = D |> normal_score_transform |> data_sphering
+    N,q = size(Z)
+    I, Q = gaussian_projection_matrix(N,q,J,k) 
+    count = 0
+    while stopping_criteria(pi,Z,I,Q,J)
+        α = optimizer(Z,J)
+        Z = structure_removal(Z,α)
+        count += 1
+        if count>=1000
+            return "Maximum iteration number ($count) reached!"
+        end
+    end
+    #print(count)
+    return Z
+end
```

Comment:
This function here that does the entire job will be our final `apply` function in the TableTransforms.jl standards. We will work iteratively to make  the implementation compliant with the interface.